### ISSUE TITLE
java.lang.IllegalArgumentException

### ISSUE THREAD

**Issue by rezafaraji93** [crisp-im/crisp-sdk-android#173]
We are facing this error:
java.lang.IllegalArgumentException - Couldn't find meta-data for provider with authority package.name.im.crisp.client.uploadfileprovider
crisp version: v2.0.0beta1

**Comment by Doc1faux** [comment]
Hi @rezafaraji93 and thank you for your feedback,

Do you have declared a `FileProvider` in your app's `AndroidManifest.xml`?

If so, Crisp SDK already declares one for attachments upload and only one could be declared per app.
So what you could do is merge both into yours as below.

#### `AndroidManifest.xml`
```xml
<!-- Adds the the Crisp SDK upload authority -->
<provider android:name="androidx.core.content.FileProvider"
  android:authorities="my.app.fileprovider;${applicationId}.im.crisp.client.uploadfileprovider"
  android:exported="false"
  android:grantUriPermissions="true"
  tools:replace="android:authorities">
  <meta-data android:name="android.support.FILE_PROVIDER_PATHS"
    android:resource="@xml/file_paths"
    tools:replace="android:resource" />
</provider>
```

#### `res/xml/file_paths.xml`
```xml
<paths>
  <files-path name="files" path="files/" />
  <cache-path name="cache" path="cache/" />

  <!-- Declares the Crisp SDK attachments path into your app's folder -->
  <files-path name="crisp_sdk_attachments" path="im.crisp.client/attachments/" />
</paths>
```

**Comment by rezafaraji93** [comment]
Thanks for your reply
I added the provider to the Manifest and now I am getting this error:
`java.lang.SecurityException: Permission Denial: starting Intent { act=android.media.action.IMAGE_CAPTURE flg=0x3 cmp=com.android.camera2/com.android.camera.CaptureActivity clip={text/uri-list hasLabel(0) {U(content)}} (has extras) } from ProcessRecord{ca7a265 7757:cash.bit24/u0a190} (pid=7757, uid=10190) with revoked permission android.permission.CAMERA`

**Comment by rezafaraji93** [comment]
I manually allowed the camera permission from settings and it didn't crash. Then, I tried to send the captured image, but after clicking on the tick icon to send, nothing happened. So now there are two problems, first, camera permission problem, Second, send capture image didn't work.

**Comment by rezafaraji93** [comment]
Yes the camera permission is mandatory in our application. When will this be fixed?

**Comment by rezafaraji93** [comment]
Would you please let me know when are you planning to update the sdk?

**Comment by Doc1faux** [comment]
Hi @rezafaraji93,

As promised, the new `2.0.1beta2` fixing your issue has just been released, I close this issue
